### PR TITLE
Add a circle

### PR DESCRIPTION
**Pull request by ecavallo**
Feel free to send this to the moon if you don't want to accumulate the cruft right now, I just wanted to try adding something.

### CODE REVIEW

**Comment by jonsterling** [comment]
holy crap that was quick!!

**Comment by ecavallo** [comment]
90% is copy-pasted from nat :sunglasses:

File: test/circle.cooltt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    i =>
+    hcom circle 0 1 {∂ i} {k _ =>
+      [ k=0 => loopn i
+      | i=0 => base
+      | i=1 => loop k
+      ]
+    }
+  ]
+
+normalize {loopn 13} -- the biggest one it can compute before Stack Overflow lol
```
Comment (jonsterling) [inline test/circle.cooltt:17]: 😎

Comment (ecavallo) [reply test/circle.cooltt:17]: Oh lol I forgot to commit my test file but you pretty much replicated it

Comment (jonsterling) [reply test/circle.cooltt:17]: Hahaha

Comment (jonsterling) [reply test/circle.cooltt:17]: I was like "Wow Evan is so amazing, he even managed to code this without any examples ??"

**Comment by favonia** [comment]
Okay this looks very high-tech. I think the fact that it overflows means that there are cases where we have to roll back and forth a long way to check multiple equalities between dimension expressions that are grouped together.

File: src/basis/PersistentTable.ml [hunk 2]
```diff
@@ -34,20 +34,25 @@ struct
...
-      reroot t';
-      match !t' with
-      | Tbl a as n ->
-        let ov' = Hashtbl.find_opt a k in
-        raw_set_opt a k ov;
-        t := n;
-        t' := Diff (k, ov', t)
-      | _ ->
-        raise Fatal
+      rerootk t' @@ fun () ->
```
Comment (favonia) [inline src/basis/PersistentTable.ml:41]: Let's put some tailcall here to make sure. ;-)

Comment (jonsterling) [reply src/basis/PersistentTable.ml:41]: feel free

File: src/lib/Lex.mll [hunk 3]
```diff
@@ -25,6 +25,7 @@ let keywords =
     ("base", BASE);
     ("loop", LOOP);
     ("circle", CIRCLE);
+    ("🍪", CIRCLE);
```
Comment (jonsterling) [inline src/lib/Lex.mll:28]: LOL

**Pull request merged**